chat_room_sso_functions.php

```diff
@@ -0,0 +1,180 @@
+#parmeter: Param $user an array containing information about users, who need to be deleted. The array user will contain comma seperated username or email, which will be used to build URL query string to delete user(s) from the Website Toolbox chat room. 
+#URL with all parameter from $user array passed in doHTTPCall function to create a request using curl or file and getting response from the Website Toolbox chat room.
+#return: Parse and return user deletion response status. 
+function userDeletionFromChatRoom($user) {
+	foreach ($user as $key => $value) {
+	  if ($value === NULL)
+		 $user[$key] = '';
+	}
+	# Generating a URL-encoded query string from the $user array.	
+	$parameters = http_build_query($user, NULL, '&');   
+	$URL = "/sso/user/delete?apikey=".API_KEY."&".$parameters;
+	# making a request using curl or file and getting response from the Website Toolbox.
+	$response = doHTTPCall($URL);
+	$response_json = json_decode($response, true);
+	# returning sso register response
+	return $response_json['success'];		  
+}
+
+#Purpose: Function for setting the password of the user for the Website Toolbox chat room. This password can be used by the user for direct login to the Website Toolbox chat room. 
+#parmeter: Param $user an array containing information about user. The array user will contain user's username/email and password, which will be used to build URL query string to set the password for the user at the Website Toolbox chat room. 
+#URL with all parameter from $user array passed in doHTTPCall function to create a request using curl or file and getting response from the Website Toolbox chat room.
+#return: Parse and return response status. 
+function chatRoomSetPassword($user) {
+	foreach ($user as $key => $value) {
+	  if ($value === NULL)
+		 $user[$key] = '';
+	}
+	# Generating a URL-encoded query string from the $user array.	
+	$parameters = http_build_query($user, NULL, '&');   
+	$URL = "/sso/user/setPassword?apikey=".API_KEY."&".$parameters;
+	# making a request using curl or file and getting response from the Website Toolbox.
+	$response = doHTTPCall($URL);
+	$response_json = json_decode($response, true);
+	# returning sso register response
+	return $response_json['success'];		  
+}
+
+
+#Purpose: Create a request using curl or file and getting response from the Website Toolbox chat room.
+#parmeter: request URL which will use to make curl request to the Website Toolbox chat room.
+#return: return response from the Website Toolbox chat room.
+function doHTTPCall($URL){
+	if (_checkBasicFunctions("curl_init,curl_setopt,curl_exec,curl_close")) {
+		$ch = curl_init("http://".HOST.$URL);
+		curl_setopt($ch, CURLOPT_HEADER, 0);
+		curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+		curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+		$response = curl_exec($ch);      
+		curl_close($ch);
+	} else if (_checkBasicFunctions("fsockopen,fputs,feof,fread,fgets,fclose")) {
+		$fsock = fsockopen(HOST, 80, $errno, $errstr, 30);
```

Comment:
As per Masuhir's latest comments, this code doesn't work. Only the curl code works. So remove this code.